miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -383,13 +389,24 @@ export.validate_storage_offsets
         end
     else
         while.true
-            # get storage offset from memory
-            dup exec.get_procedure_storage_offset
-            # => [storage_offset, index, num_storage_slots, num_account_procedures]
+            # get storage offset and size from memory
+            dup exec.get_procedure_metadata
+            # => [storage_offset, storage_size, index, num_storage_slots, num_account_procedures]
 
             # assert that storage offset is in bounds
-            dup.2 lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
-            # => [index, num_storage_slots, num_account_procedures]
+            dup dup.4 lt assert.err=ERR_STORAGE_SLOT_INDEX_OUT_OF_BOUNDS
+            # => [storage_offset, storage_size, index, num_storage_slots, num_account_procedures]
+
+            # assert that if size is 0 then offset is 0
+            dup.1 eq.0
+            if.true
+                dup eq.0 assert.err=ERR_INVALID_STORAGE_OFFSET_FOR_SIZE
+            end
+            # => [storage_offset, storage_size, index, num_storage_slots, num_account_procedures]
```

Comment:
Let's add a `TODO` for this block to figure out how to do this without the `if` statement.

Reply:
Why would we need to remove the `if` statement?

Reply:
`if` statements are relatively expensive. So, if we can do this with some small number (e.g., half a dozen or so) non-conditional instructions, this would be quite a bit more efficient.